src/Provider/Doctrine/FieldDef.php

```diff
@@ -78,4 +78,35 @@ public static function reference($name)
             return $factory->get($name);
         };
     }
+
+    /**
+     * Defines a field to `get()` a collection of named entities from the factory.
+     *
+     * The normal semantics of `get()` apply.
+     * 
+     * Normally this means that the field gets a fresh instance of the named
+     * entity. If a singleton has been defined, a collection with a single instance will be returned.
+     *
+     * @param string $name
+     * @param int $numberOfInstances
+     *
+     * @throws \InvalidArgumentException
+     * @return callable
+     */
+    public static function references($name, $numberOfInstances = 3)
+    {
+        if ($numberOfInstances < 1) {
+            throw new \InvalidArgumentException('Can only get >= 1 instances');
+        }
+
+        return function(FixtureFactory $factory) use ($name, $numberOfInstances) {
+            if ($numberOfInstances > 1 && $factory->isSingleton($name)) {
+                $numberOfInstances = 1;
+            }
```

Comment:
I see where this is coming from. In general if you have a list of elements you don't want the same item to be in the list multiple times.

I feel like having ``$numberOfInstances`` set to 1 by default somehow removes the need for this check. Because then the only way the items is added multiple times is if the library user explicitly asks for 3 instances.

If we actually want this check then I am not sure whether it wouldn't make more sense to move this into the ``FixtureFactory::getList`` method.

Thoughts?